HiringApp/HiringAppCore/Providers/DBProvider.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  DBProvider.swift
+//  Pods
+//
+//  Created by Alba Luján on 23/6/17.
+//
+//
+
+import Foundation
+import RealmSwift
+import Deferred
+
+protocol DBProviderType {
+    func write(tech: TechnologyRealm) -> Task<()>
+    func read() -> [TechnologyRealm]?
+}
+
+class DBProvider: DBProviderType {
+    
+    var realm: Realm?
```

Comment:
according to the realm docs : 

To handle errors when first accessing a Realm on a given thread, use Swift’s built-in error handling mechanism:

do {
  let realm = try Realm()
} catch let error as NSError {
  // handle error
}